sui_core/src/unit_tests/consensus_tests.rs

```diff
@@ -150,13 +150,11 @@ async fn handle_consensus_output() {
         .await
         .unwrap();
 
-    // Wait for the certificate to be processed and ensure the last consensus index
-    // has been updated.
-    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
-    assert_eq!(
-        state.db().last_consensus_index().unwrap(),
-        SequenceNumber::from(1)
-    );
+    // Wait for the certificate to be processed and ensure the last consensus index is correctly updated.
+    // (We need to wait on storage for that.)
+    while state.db().last_consensus_index().unwrap() != SequenceNumber::from(1) {
+        tokio::time::sleep(std::time::Duration::from_millis(10)).await;
+    }
```

Comment:
Is there a better way to do that actually? Something like the `notify_read` we had in Narwhal perhaps